aws/rosa-hcp-dual-region/camunda-version/8.7/s3-elastic-backup/backup-bucket.tf

```diff
@@ -0,0 +1,135 @@
+variable "bucket_name" {
+  type        = string
+  description = "Name of the bucket used to backup the platform"
+  default     = "camunda-elastic-backup-rosa-dual"
+}
+
+resource "aws_s3_bucket" "elastic_backup" {
+  bucket = var.bucket_name
+
+  tags = {
+    Name = var.bucket_name
+  }
+}
+
+resource "aws_kms_key" "backup_bucket_key" {
+  description             = "This key is used to encrypt bucket ${var.bucket_name}"
+  deletion_window_in_days = 10
+  enable_key_rotation     = true
+}
+
+resource "aws_s3_bucket_server_side_encryption_configuration" "encrypt_bucket" {
+  bucket = aws_s3_bucket.elastic_backup.id
+
+  rule {
+    apply_server_side_encryption_by_default {
+      kms_master_key_id = aws_kms_key.backup_bucket_key.arn
+      sse_algorithm     = "aws:kms"
+    }
+  }
+}
+
+resource "aws_s3_bucket_server_side_encryption_configuration" "encrypt_log_bucket" {
+  bucket = aws_s3_bucket.log_bucket.id
+
+  rule {
+    apply_server_side_encryption_by_default {
+      kms_master_key_id = aws_kms_key.backup_bucket_key.arn
+      sse_algorithm     = "aws:kms"
+    }
+  }
+}
+
+
+resource "aws_s3_bucket_versioning" "versionning_backup" {
+  bucket = aws_s3_bucket.elastic_backup.id
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+resource "aws_s3_bucket_versioning" "versionning_logs" {
+  bucket = aws_s3_bucket.log_bucket.id
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+resource "aws_s3_bucket_logging" "this" {
+  bucket        = aws_s3_bucket.elastic_backup.id
+  target_bucket = aws_s3_bucket.log_bucket.id
+  target_prefix = "log/"
+}
+
+resource "aws_s3_bucket" "log_bucket" {
+  bucket = "${var.bucket_name}-log"
+}
+
+resource "aws_s3_bucket_acl" "log_bucket" {
+  acl    = "log-delivery-write"
+  bucket = aws_s3_bucket.log_bucket.id
+}
```

Comment:
as mentioned in the guide, the acl was causing problems.
I removed it in the end to continue